crates/parcel_core/src/types/asset.rs

```diff
@@ -92,11 +118,42 @@ impl Asset {
 
     hasher.finish()
   }
+
+  /// Build a new empty asset
+  pub fn new_empty(file_path: PathBuf, source_code: Rc<SourceCode>) -> Self {
+    let asset_type =
+      FileType::from_extension(file_path.extension().and_then(|s| s.to_str()).unwrap_or(""));
+
+    // TODO: rest of this
+    Self {
```

Comment:
/nit If you derive from default, you can make this shorter by using `..Asset::default()`